Microsoft.Azure.Cosmos/src/Handler/TelemetryHandler.cs

```diff
@@ -48,10 +51,42 @@ public TelemetryHandler(ClientTelemetry telemetry)
                     DefaultTrace.TraceError("Error while collecting telemetry information : {0}", ex);
                 }
             }
+
             return response;
         }
 
-        private bool IsAllowed(RequestMessage request)
+        /// <summary>
+        /// Check if Collection should happen or not, if yes then return the client job instance where information needs to send
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="clientTelemetryJob"></param>
+        /// <returns>true/false</returns>
+        private bool IsEligibleForTelemetryCollection(RequestMessage request, out ClientTelemetry clientTelemetryJob)
+        {
+            return this.IsClientTelemetryJobRunning(out clientTelemetryJob) && this.IsRequestAllowed(request);
+        }
+
+        /// <summary>
+        /// Check if Client Telemetry Job is running in background.
+        /// </summary>
+        /// <param name="clientTelemetryJob"></param>
+        /// <returns>true/false</returns>
+        private bool IsClientTelemetryJobRunning(out ClientTelemetry clientTelemetryJob)
+        {
+            clientTelemetryJob = this.Client.DocumentClient.ClientTelemetryInstance;
+            if (this.Client.DocumentClient.ClientTelemetryInstance == null)
```

Comment:
Why are you checking `this.Client.DocumentClient.ClientTelemetryInstance`? If you just snapshoted on `clientTelemetryJob` shouldn't you be able to use that?

Also, this could be simplified?

```suggestion
           return clientTelemetryJob != null;
```